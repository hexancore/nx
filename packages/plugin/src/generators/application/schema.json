{
  "$schema": "https://json-schema.org/schema",
  "$id": "Application",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use?"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the application is placed.",
      "x-priority": "important",
      "$default": {
        "$source": "argv",
        "index": 0
      },
    },
    "type": {
      "description": "The application type.",
      "type": "string",
      "enum": ["backend"],
      "x-prompt": "Which application type would you like to use ?",
      "x-priority": "important"
    }
  },
  "required": ["name", "type"]
}
