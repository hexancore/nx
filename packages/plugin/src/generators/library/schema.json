{
  "$schema": "https://json-schema.org/schema",
  "$id": "Library",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use?"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the lib is placed.",
      "x-priority": "important",
      "$default": {
        "$source": "argv",
        "index": 0
      },
    },
    "type": {
      "description": "The library type.",
      "type": "string",
      "enum": ["backend", "shared"],
      "x-prompt": "Which library type would you like to use ? Choose 'none' to skip build setup.",
      "x-priority": "important"
    }
  },
  "required": ["name", "type"]
}
