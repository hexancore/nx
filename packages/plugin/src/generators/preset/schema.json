{
  "$schema": "https://json-schema.org/draft/2020-12/output/schema",
  "$id": "Preset",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use?"
    }
  },
  "required": ["name"]
}
