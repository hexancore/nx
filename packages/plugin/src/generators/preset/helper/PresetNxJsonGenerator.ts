import {
  Tree,
  updateJson
} from '@nx/devkit';

export class PresetNxJsonGenerator {

  public static run(tree: Tree) {
    const g = new this();
    g.apply(tree);
  }

  public apply(tree: Tree): void {
    updateJson(tree, 'nx.json', (o) => {
      o.extends = 'nx/presets/npm.json';
      o.pluginsConfig = {
        "@nx/js": {
          "analyzeSourceFiles": true
        }
      };

      o.namedInputs = o.namedInputs ?? {};
      o.namedInputs = {
        "default": [
          "{projectRoot}/**/*",
          "sharedGlobals"
        ],
        "sharedGlobals": [],
      };
      o.namedInputs.production = [
        "default",
        "!{projectRoot}/jest.config.{js,ts,mjs,mts}",
        "!{projectRoot}/vite.config.{js,ts,mjs,mts}",
        "!{projectRoot}/.eslintrc.json"
      ];

      o.targetDefaults = {
        "build": {
          "inputs": ["default", "!{projectRoot}/test/**/*"],
          "dependsOn": ["^build"],
          "cache": true
        },
        "lint": {
          "inputs": [
            "default",
            "{workspaceRoot}/.eslintrc.json",
            "{workspaceRoot}/.eslintignore"
          ],
          "cache": true
        },
        "lint-fix": {
          "cache": false
        },
        "test": {
          "inputs": [
            "default",
            "^default",
            "{workspaceRoot}/jest.preset.js"
          ],
          "cache": true
        },
        "test-watch": {
          "inputs": ["default", "^default", "{workspaceRoot}/jest.preset.js"],
          "cache": true
        },
        "@nx/vite:test": {
          "cache": true,
          "inputs": [
            "default",
            "^default"
          ]
        },
        "@nx/eslint:lint": {
          "inputs": [
            "default",
            "{workspaceRoot}/.eslintrc.json",
            "{workspaceRoot}/.eslintignore"
          ],
          "cache": true
        }
      };

      o.generators = {
        "@nx/workspace:move": {
          "projectNameAndRootFormat": "as-provided"
        }
      };
      o.useInferencePlugins = false;
      o.defaultBase = 'main';

      o['workspaceLayout'] = {
        "appsDir": "apps",
        "libsDir": "libs"
      };

      return o;
    });
  }
}