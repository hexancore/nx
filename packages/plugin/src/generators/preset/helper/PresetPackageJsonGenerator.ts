import {
  addDependenciesToPackageJson,
  updateJson,
  Tree,
} from '@nx/devkit';

export class PresetPackageJsonGenerator {

  public constructor(
    private nxVersion: string,
    private hcNxPluginVersion: string
  ) {

  }

  public static run(tree: Tree, options: {
    nxVersion: string,
    hcNxPluginVersion: string
  }) {
    const g = new this(options.nxVersion, options.hcNxPluginVersion);
    g.apply(tree);
  }

  public apply(tree: Tree): void {
    this.generateDeps(tree);

    updateJson(tree, 'package.json', (o) => {
      o.license = 'UNLICENSED';
      o.scripts = o.scripts ?? {};

      o.scripts['prepare'] = "node .husky/install.mjs",
        o.scripts['precommit'] = "nx run-many --targets=lint,test,build";
      o.scripts['ci'] = "nx run-many --targets=lint,test,build,e2e --configuration=ci";
      return o;
    });
  }

  private generateDeps(tree: Tree): void {
    const deps = this.createDeps();
    const devDeps = this.createDevDeps();
    addDependenciesToPackageJson(tree, deps, devDeps);
  }

  private createDeps(): Record<string, string> {
    const hexancore = {
      '@hexancore/core': '0.14.*',
      '@hexancore/common': '0.13.*',
      '@hexancore/cloud': '0.1.*',
      '@hexancore/auth': '0.3.*',
      '@hexancore/typeorm': '0.14.*',
    };

    return {
      ...hexancore,
      'tslib': '2.6.2'
    };
  }

  private createDevDeps(): Record<string, string> {
    return {
      ...this.createHexancoreDevDeps(),
      ...this.createNxDevDeps(),
      ...this.createFrontendDevDeps(),
      ...this.createJestDevDeps(),
      ...this.createEslintDevDeps(),
      ...this.createEssentialsDevDeps()
    };
  }

  private createNxDevDeps(): Record<string, string> {
    return {
      "@nx/devkit": this.nxVersion,
      "@nx/esbuild": this.nxVersion,
      "@nx/eslint-plugin": this.nxVersion,
      "@nx/jest": this.nxVersion,
      "@nx/js": this.nxVersion,
      "@nx/linter": this.nxVersion,
      "@nx/plugin": this.nxVersion,
    };
  }

  private createHexancoreDevDeps(): Record<string, string> {
    const [major, minor] = this.hcNxPluginVersion.split('.');

    return {
      "@hexancore/nx": `${major}.${minor}.*`,
      "@hexancore/mocker": "^1.1.2",
    };
  }

  private createFrontendDevDeps(): Record<string, string> {
    const vue = {
      "@vue/eslint-config-typescript": "^11.0.3",
      "@vue/test-utils": "^2.4.5",
      "@vue/tsconfig": "^0.4.0",
      "vue": "^3.4.25",
      "vue-eslint-parser": "^9.4.2",
      "vue-router": "^4.3.2",
      "pinia": "^2.1.7",
      "vue-tsc": "^1.8.27",
      "vee-validate": "^4.12.6",
    };

    const vueI18n = {
      "vue-i18n": "^9.13.1",
      "@intlify/eslint-plugin-vue-i18n": "^2.0.0",
      "@intlify/unplugin-vue-i18n": "^2.0.0",
      "unplugin-swc": "^1.4.5",
    };

    const vite = {
      "@vitejs/plugin-basic-ssl": "^1.1.0",
      "@vitejs/plugin-vue": "^5.0.4",
      "@vitest/coverage-v8": "1.6.*",
      "@vitest/ui": "1.6.*",
      "vitest": "1.6.*",
      "@rollup/plugin-node-resolve": "^15.2.3",
      "rollup": "^4.17.0",
      "vite": "^5.2.10",
      "vite-tsconfig-paths": "^4.3.2",
      "autoprefixer": "^10.4.19",
      "@babel/core": "^7.24.4",
      "@babel/plugin-transform-modules-commonjs": "^7.24.1",
      "@babel/preset-env": "^7.24.4",
    };

    const primevue = {
      "primeflex": "^3.3.1",
      "primeicons": "^6.0.1",
      "primevue": "^3.52.0",
    };

    const style = {
      "postcss": "^8.4.38",
      "sass": "^1.75.0",
      "stylelint": "^15.11.0",
      "stylelint-config-standard": "^33.0.0",
      "@samatech/postcss-basics": "^0.6.0",
    };

    return {
      ...vue,
      ...vueI18n,
      ...vite,
      ...primevue,
      ...style
    };
  }

  public createEslintDevDeps(): Record<string, string> {
    return {
      "eslint": "8.57.0",
      "eslint-plugin-import": "^2.29.1",
      "eslint-plugin-vue": "^9.25.0",
      "@typescript-eslint/parser": "7.3.0",
      "@stylistic/eslint-plugin": "^2.1.0",
    };
  }

  public createJestDevDeps(): Record<string, string> {
    return {
      "happy-dom": "^9.20.3",
      "jest": "^29.7.0",
      "jest-docblock": "^29.7.0",
      "jest-environment-jsdom": "^29.7.0",
      "jest-environment-node": "^29.7.0",
      "jest-expect-message": "^1.1.3",
      "jest-runner": "^29.7.0",
      "jest-runner-groups": "^2.2.0",
      "jsdom": "~22.1.0",
      "ts-jest": "^29.1.2",
      "@types/jest": "^29.5.12",
      "@jest/globals": "^29.7.0",
    };
  }

  public createEssentialsDevDeps(): Record<string, string> {
    return {
      "tsconfig": "^7.0.0",
      "typescript": "5.4.5",
      "typescript-transform-paths": "^3.4.7",
      "esbuild": "^0.19.12",
      "axios-mock-adapter": "^1.22.0",
      "@types/node": "20.12.12",
      "@nestjs/testing": "^10.3.8",
      "husky": "^9.0.11",
    };
  }
}