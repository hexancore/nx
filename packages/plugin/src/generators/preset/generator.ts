import {
  Tree,
  formatFiles,
  generateFiles
} from '@nx/devkit';
import * as path from 'path';
import { HcNxHelper } from '../../util/NxHelper';
import { PresetNxJsonGenerator } from './helper/PresetNxJsonGenerator';
import { PresetPackageJsonGenerator } from './helper/PresetPackageJsonGenerator';
import { PresetGeneratorSchema } from './schema';
import { readFileSync } from 'fs';

export async function presetGenerator(
  tree: Tree,
  options: PresetGeneratorSchema
) {

  const nxVersion = HcNxHelper.getPluginNxVersion();
  const hcNxPluginVersion = HcNxHelper.getVersion();

  PresetPackageJsonGenerator.run(tree, { nxVersion, hcNxPluginVersion });
  PresetNxJsonGenerator.run(tree);

  tree.delete('.prettierrc');
  const templateContext = {
    workspace: {
      name: options.name,
      nxVersion
    }
  };
  const filesPath = path.join(__dirname, 'files');
  tree.write('.vscode/extensions.json', readFileSync(path.join(filesPath, 'vscode/extensions.json')));
  tree.write('.vscode/settings.json.template', readFileSync(path.join(filesPath, 'vscode/settings.json')));
  generateFiles(tree, path.join(filesPath, './.husky'), './.husky', templateContext);
  generateFiles(tree, path.join(filesPath, 'workspace'), './', templateContext);

  await formatFiles(tree);
}

export default presetGenerator;
