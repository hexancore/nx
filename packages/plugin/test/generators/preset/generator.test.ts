import { Tree } from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';

import { presetGenerator } from '../../../src/generators/preset/generator';
import { PresetGeneratorSchema } from '../../../src/generators/preset/schema';
import { exceptWorkspaceFileMatchSnapshot } from '../../helper/functions';

describe('preset generator', () => {
  let tree: Tree;

  const options: PresetGeneratorSchema = { name: 'test-workspace' };
  beforeAll(async () => {
    tree = createTreeWithEmptyWorkspace();
    await presetGenerator(tree, options);
  });

  const expectedWorkespaceFilesToMatchSnapshots = [
    'nx.json',
    'package.json',
    '.env',
    '.env.development',
    '.env.test',
    '.npmrc',
    'CONTRIBUTING.md',
    'README.md',
    'docker/app/Dockerfile',
    'docker/app/pm2.config.cjs',
    'docker/dev/.env',
    'Makefile',
    'tsconfig.base.json',

    'docker-compose-dev.yaml',
  ];
  test.each(expectedWorkespaceFilesToMatchSnapshots.map(i => [i]))('%s should match snapshot', (filePath) => {
    exceptWorkspaceFileMatchSnapshot(tree, filePath);
  });

  test('only file from list should exist', () => {
    const expectedWorkspaceFilesToExists = [
      '.eslintignore',
      '.eslintrc.json',
      '.prettierrc',
      '.gitignore',
      '.gitattributes',
      '.editorconfig',
      '.dockerignore',
      '.vscode/extensions.json',
      '.vscode/settings.json.template',
      '.husky/install.mjs',
      '.husky/pre-commit',
      'jest.config.ts',
      'jest.preset.ts',
      'bin/util/MakeHelp',
      'bin/util/util.sh',
      'docker/dev/ca/ca.crt',
      'docker/dev/ca/ca.key',
      'docker/dev/ca/ca.srl',
      'docker/dev/ca/certs/db/db.crt',
      'docker/dev/ca/certs/db/db.key',
      'docker/dev/ca/certs/redis/redis.crt',
      'docker/dev/ca/certs/redis/redis.key',
      'docker/dev/ca/create_ca.sh',
      'docker/dev/ca/gen_cert.sh',
      'docker/dev/redis/users.acl',
      ...expectedWorkespaceFilesToMatchSnapshots
    ];
    const diff = tree.listChanges().map(c => c.path).filter(x => !expectedWorkspaceFilesToExists.includes(x));
    expect(diff).toEqual([]);
  });

});
