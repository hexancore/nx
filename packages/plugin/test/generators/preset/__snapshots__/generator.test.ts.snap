// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preset generator .env should match snapshot 1`] = `
"PROJECT=test-workspace
TZ=Etc/UTC
APP_HOME=./
"
`;

exports[`preset generator .env.development should match snapshot 1`] = `
"APP_ENV=dev
NODE_TLS_REJECT_UNAUTHORIZED=0
#DEBUG=ioredis:*
"
`;

exports[`preset generator .env.test should match snapshot 1`] = `
"APP_ENV=test
NODE_TLS_REJECT_UNAUTHORIZED=0"
`;

exports[`preset generator .npmrc should match snapshot 1`] = `
"ignore-workspace-root-check=true
strict-peer-dependencies=false
auto-install-peers=true
"
`;

exports[`preset generator CONTRIBUTING.md should match snapshot 1`] = `
"# Contributing to test-workspace

First off, thanks for taking the time to contribute! ðŸŽ‰

The following is a set of guidelines for contributing to test-workspace. These are mostly guidelines, not rules. Use your best judgment, and feel free to propose changes to this document in a pull request.

## How Can I Contribute?

### Reporting Bugs

This section guides you through submitting a bug report for test-workspace. Following these guidelines helps maintainers understand your report, reproduce the behavior, and find related reports.

- **Ensure the bug was not already reported** by searching on [PLACEHOLDER]
- If you're unable to find an open issue addressing the problem, open a new one. Be sure to include:
  - A clear and descriptive title.
  - A detailed description of the bug.
  - Steps to reproduce the bug.
  - Expected and actual results.
  - Screenshots or code samples, if applicable.

### Suggesting Enhancements

This section guides you through submitting an enhancement suggestion for test-workspace, including completely new features and minor improvements to existing functionality.

- **Ensure the enhancement was not already suggested** by searching on [PLACEHOLDER].
- If you're unable to find an open issue addressing the idea, open a new one. Be sure to include:
  - A clear and descriptive title.
  - A detailed description of the enhancement.
  - Any relevant screenshots or mockups.

### Your First Code Contribution

### Pull Requests

The process described here will help you contribute to test-workspace. Please follow these steps to ensure your contribution is accepted:

1. Fork the repository and create your branch from \`main\`.
2. If you've added code that should be tested, add tests.
3. If you've changed APIs, update the documentation.
4. Ensure the test suite passes.
5. Make sure your code lints.
6. Issue that pull request!

### Coding Style

Please follow the coding standards used throughout the existing code base. Some general guidelines:

- Use descriptive variable and function names.
- Write comments where necessary for clarity.
- Ensure your code passes all linting checks.

## Style Guides

### Git Commit Messages

- Use the present tense ("Add feature" not "Added feature").
- Use the imperative mood ("Move cursor to..." not "Moves cursor to...").
- Limit the first line to 72 characters or less.
- Reference issues and pull requests liberally.

## Additional Notes

### Issue and Pull Request Labels

This section lists the labels we use to help us track and manage issues and pull requests.

- \`bug\`: Something isn't working
- \`enhancement\`: New feature or request
- \`question\`: Further information is requested
- \`documentation\`: Improvements or additions to documentation
- \`help wanted\`: Extra attention is needed

---

Thank you for your contributions! Your help is greatly appreciated.
"
`;

exports[`preset generator Makefile should match snapshot 1`] = `
"export ENV ?= dev
export WORKSPACE = test-workspace

## PATHS
export BIN_ROOT_PATH = ./bin
export UTIL_ROOT_PATH = $(realpath $(BIN_ROOT_PATH)/util)
export UTIL_PATH = $(UTIL_ROOT_PATH)/util.sh

#ifndef CI
## dev.env
	#include dev.env
	#export $(shell sed 's/=.*//' dev.env)
#endif

## VERSION
ifeq "$(ENV)" "dev"
  export VERSION = dev
endif

DC=docker compose -p $(WORKSPACE)

# Help target settings
INFO = Showing targets for all of ENV(default: dev)
EXTRA_MAKE_ARGS = ENV=test|prod
HELP_TARGET_MAX_CHAR_NUM = 30
HAS_DEPS = yes
.DEFAULT_GOAL := help
include $(UTIL_ROOT_PATH)/MakeHelp

# Check version is sets and is not help target
ifneq ($(MAKECMDGOALS),)
  ifneq ($(MAKECMDGOALS),help)
    ifndef VERSION
      $(error VERSION is not set)
    endif
  endif
endif

ifeq "$(ENV)" "prod"
## prod targets
endif

ifeq "$(ENV)" "dev"

init_devenv:
	cp ./.vscode/settings.json.template ./.vscode/settings.json

## Reinitialize testing env
up:
	if [ "$CI" = "1" ]; then
		mkdir -p ./tmp/docker
		chmod 777 ./tmp/docker
		chmod 640 -R ./docker/ca/certs/db
	fi
	$(DC) -f docker-compose-dev.yaml --env-file ./docker/dev/.env up -d --wait --wait-timeout 5
	sleep 3

## Clear testing env
down:
	$(DC) down -t 2

build_dev_image:
	DOCKER_BUILDKIT=1 docker build -t test-workspace/dev:latest -f ./docker/app/Dockerfile --target=dev .
endif"
`;

exports[`preset generator README.md should match snapshot 1`] = `
"# test-workspace

Replace with Yours project workspace description.

## Getting started

To start contributing you need install some tools and init developer environment.

### Tools

- NodeJS
- pnpm
- Nx (\`pnpm add --global nx@19.1.0\`)
- Docker + Compose
- make

To init developer environment run:

\`\`\`bash
make init_devenv
\`\`\`

## Contributing

Contributing Guide is in separate file [CONTRIBUTING.md](./CONTRIBUTING.md)
"
`;

exports[`preset generator docker/app/Dockerfile should match snapshot 1`] = `
"# pnpm https://github.com/moby/moby/issues/40449#issuecomment-1758956399
ARG NODE_IMAGE=node:22-alpine3.18
ARG PM2_VERSION=5.3.1
ARG USER_UID=1000

FROM "$NODE_IMAGE" as deps

ENV NODE_ENV=production
ENV PNPM_HOME="/app/.pnpm-store"
ENV PATH="$PNPM_HOME:$PATH"

RUN mkdir -p /app/.pnpm-store && chown -R node:node /app && chmod -R 777 /app && corepack enable && pnpm config set store-dir "$PNPM_HOME"

USER node
WORKDIR /app

COPY --chown=node:node ./pnpm-lock.yaml ./
RUN --mount=type=cache,target=/app/.pnpm-store,uid=\${USER_UID},gid=\${USER_UID} pnpm fetch --prod --ignore-scripts

COPY --chown=node:node ./pnpm-lock.yaml ./package.json ./prod/
RUN --mount=type=cache,ro,target=/app/.pnpm-store,uid=\${USER_UID},gid=\${USER_UID} cd prod && pnpm install --prod --offline --frozen-lockfile --ignore-scripts

##########################
# DEV container tool stage
FROM deps as dev
RUN --mount=type=cache,target=/app/.pnpm-store,uid=\${USER_UID},gid=\${USER_UID} pnpm fetch --dev --ignore-scripts

COPY --chown=node:node ./package.json ./
RUN --mount=type=cache,ro,target=/app/.pnpm-store,uid=\${USER_UID},gid=\${USER_UID} pnpm install --prod=false --offline --frozen-lockfile --ignore-scripts

EXPOSE 3000

CMD ["sleep", "1000"]

#######################
# API application stage
FROM $NODE_IMAGE as app_api
ARG APP_ID

RUN mkdir /app && chown node:node /app && chmod 770 /app && npm i -g pm2@\${PM2_VERSION}
WORKDIR /app

RUN --mount=type=bind,from=deps,source=/app,target=/tmp/deps cp -R -p /tmp/deps/prod/* ./

COPY --chown=node:node ./dist/apps/\${APP_ID} ./apps/\${APP_ID}
COPY --chown=node:node ./dist/libs ./libs
COPY --chown=node:node ./docker/app/pm2.config.cjs ./pm2.config.cjs

USER node
ENV NODE_ENV=production
EXPOSE 3000
RUN ln -s apps/\${APP_ID}/src/index.js entrypoint

CMD ["pm2-runtime", "--json", "pm2.config.cjs"]
"
`;

exports[`preset generator docker/app/pm2.config.cjs should match snapshot 1`] = `
"var fs = require('fs');
const configPath = process.env['PM2_CONFIG_PATH'] ?? './config/pm2.config.json';
const config = JSON.parse(fs.readFileSync(configPath, { encoding: 'utf8' }));
module.exports = {
  apps: [
    {
      script: './entrypoint',
      ...config,
    },
  ],
};
"
`;

exports[`preset generator docker/dev/.env should match snapshot 1`] = `
"WORKSPACE=test-workspace

HOST_DATA_DIR=./tmp/docker
HOST_CONFIG_DIR=./docker/dev
HOST_CA_DIR=./docker/dev/ca
HOST_CERTS_DIR=./docker/dev/ca/certs
HOST_CA_CRT=./docker/dev/ca/ca.crt

# SERVICES

# Redis
REDIS_HOST_PORT=25010
REDIS_PASSWORD=testredis

# DB
DB_HOST_PORT=25011

"
`;

exports[`preset generator docker-compose-dev.yaml should match snapshot 1`] = `
"version: '3.6'
services:
  redis:
    image: bitnami/redis:7.2
    container_name: \${PROJECT}-redis
    restart: always
    environment:
      - ALLOW_EMPTY_PASSWORD=no
      - REDIS_AOF_ENABLED=no
      - REDIS_PASSWORD=\${REDIS_PASSWORD}
      - REDIS_ACLFILE=/bitnami/redis/config/users.acl
    ports:
      - \${REDIS_HOST_PORT}:6379
    volumes:
      - \${HOST_CONFIG_DIR}/redis:/bitnami/redis/config
    tmpfs:
      - /bitnami/redis/data:size=64M
  db:
    image: bitnami/postgresql:16.2.0
    container_name: \${PROJECT}-db
    restart: always
    environment:
      - POSTGRESQL_POSTGRES_PASSWORD=testroot
      - POSTGRESQL_DATABASE=\${PROJECT}
      - POSTGRESQL_USERNAME=\${PROJECT}
      - POSTGRESQL_PASSWORD=test
      - POSTGRESQL_ENABLE_TLS=yes
      - POSTGRESQL_TLS_CERT_FILE=/opt/bitnami/postgresql/certs/db.crt
      - POSTGRESQL_TLS_KEY_FILE=/opt/bitnami/postgresql/certs/db.key
      - POSTGRESQL_TLS_CA_FILE=/opt/bitnami/postgresql/certs/ca.crt
    volumes:
      - \${HOST_CERTS_DIR}/db/db.crt:/opt/bitnami/postgresql/certs/db.crt
      - \${HOST_CERTS_DIR}/db/db.key:/opt/bitnami/postgresql/certs/db.key
      - \${HOST_CA_CRT}:/opt/bitnami/postgresql/certs/ca.crt
    mem_limit: 512M
    tmpfs:
      - /tmp:size=256M
    ports:
      - \${DB_HOST_PORT}:5432
"
`;

exports[`preset generator nx.json should match snapshot 1`] = `
"{
  "affected": {
    "defaultBase": "main"
  },
  "targetDefaults": {
    "build": {
      "inputs": ["default", "!{projectRoot}/test/**/*"],
      "dependsOn": ["^build"],
      "cache": true
    },
    "lint": {
      "inputs": [
        "default",
        "{workspaceRoot}/.eslintrc.json",
        "{workspaceRoot}/.eslintignore"
      ],
      "cache": true
    },
    "lint-fix": {
      "cache": false
    },
    "test": {
      "inputs": ["default", "^default", "{workspaceRoot}/jest.preset.js"],
      "cache": true
    },
    "test-watch": {
      "inputs": ["default", "^default", "{workspaceRoot}/jest.preset.js"],
      "cache": true
    },
    "@nx/vite:test": {
      "cache": true,
      "inputs": ["default", "^default"]
    },
    "@nx/eslint:lint": {
      "inputs": [
        "default",
        "{workspaceRoot}/.eslintrc.json",
        "{workspaceRoot}/.eslintignore"
      ],
      "cache": true
    }
  },
  "extends": "nx/presets/npm.json",
  "pluginsConfig": {
    "@nx/js": {
      "analyzeSourceFiles": true
    }
  },
  "namedInputs": {
    "default": ["{projectRoot}/**/*", "sharedGlobals"],
    "sharedGlobals": [],
    "production": [
      "default",
      "!{projectRoot}/jest.config.{js,ts,mjs,mts}",
      "!{projectRoot}/vite.config.{js,ts,mjs,mts}",
      "!{projectRoot}/.eslintrc.json"
    ]
  },
  "generators": {
    "@nx/workspace:move": {
      "projectNameAndRootFormat": "as-provided"
    }
  },
  "useInferencePlugins": false,
  "defaultBase": "main",
  "workspaceLayout": {
    "appsDir": "apps",
    "libsDir": "libs"
  }
}
"
`;

exports[`preset generator package.json should match snapshot 1`] = `
"{
  "name": "@proj/source",
  "dependencies": {
    "@hexancore/auth": "0.3.*",
    "@hexancore/cloud": "0.1.*",
    "@hexancore/common": "0.13.*",
    "@hexancore/core": "0.14.*",
    "@hexancore/typeorm": "0.14.*",
    "fs-extra": "^11.2.0",
    "reflect-metadata": "^0.1.13",
    "tslib": "2.6.2"
  },
  "devDependencies": {
    "@babel/core": "^7.24.4",
    "@babel/plugin-transform-modules-commonjs": "^7.24.1",
    "@babel/preset-env": "^7.24.4",
    "@hexancore/mocker": "^1.1.2",
    "@hexancore/nx": "0.1.1",
    "@intlify/eslint-plugin-vue-i18n": "^2.0.0",
    "@intlify/unplugin-vue-i18n": "^4.0.0",
    "@jest/globals": "^29.7.0",
    "@nestjs/testing": "^10.3.8",
    "@nx/devkit": "19.1.0",
    "@nx/esbuild": "19.1.0",
    "@nx/eslint-plugin": "19.1.0",
    "@nx/jest": "19.1.0",
    "@nx/js": "19.1.0",
    "@nx/linter": "19.1.0",
    "@nx/plugin": "19.1.0",
    "@rollup/plugin-node-resolve": "^15.2.3",
    "@samatech/postcss-basics": "^0.6.0",
    "@stylistic/eslint-plugin": "^2.1.0",
    "@types/jest": "^29.5.12",
    "@types/node": "20.12.12",
    "@typescript-eslint/parser": "^7.10.0",
    "@vitejs/plugin-basic-ssl": "^1.1.0",
    "@vitejs/plugin-vue": "^5.0.4",
    "@vitest/coverage-v8": "1.6.*",
    "@vitest/ui": "1.6.*",
    "@vue/eslint-config-typescript": "^11.0.3",
    "@vue/test-utils": "^2.4.5",
    "@vue/tsconfig": "^0.4.0",
    "autoprefixer": "^10.4.19",
    "axios-mock-adapter": "^1.22.0",
    "esbuild": "^0.19.12",
    "eslint": "8.57.0",
    "eslint-plugin-import": "^2.29.1",
    "eslint-plugin-vue": "^9.25.0",
    "happy-dom": "^14.11.0",
    "husky": "^9.0.11",
    "jest": "^29.7.0",
    "jest-docblock": "^29.7.0",
    "jest-environment-jsdom": "^29.7.0",
    "jest-environment-node": "^29.7.0",
    "jest-expect-message": "^1.1.3",
    "jest-runner": "^29.7.0",
    "jest-runner-groups": "^2.2.0",
    "jsdom": "^24.0.0",
    "pinia": "^2.1.7",
    "postcss": "^8.4.38",
    "primeflex": "^3.3.1",
    "primeicons": "^7.0.0",
    "primevue": "^3.52.0",
    "rollup": "^4.17.0",
    "sass": "^1.75.0",
    "stylelint": "^16.6.0",
    "stylelint-config-standard": "^36.0.0",
    "ts-jest": "^29.1.2",
    "tsconfig": "^7.0.0",
    "typescript": "5.4.5",
    "typescript-transform-paths": "^3.4.7",
    "unplugin-swc": "^1.4.5",
    "vee-validate": "^4.12.6",
    "vite": "^5.2.10",
    "vite-tsconfig-paths": "^4.3.2",
    "vitest": "1.6.*",
    "vue": "^3.4.25",
    "vue-eslint-parser": "^9.4.2",
    "vue-i18n": "^9.13.1",
    "vue-router": "^4.3.2",
    "vue-tsc": "^1.8.27"
  },
  "license": "UNLICENSED",
  "scripts": {
    "prepare": "node .husky/install.mjs",
    "precommit": "nx run-many --targets=lint,test,build",
    "ci": "nx run-many --targets=lint,test,build,e2e --configuration=ci"
  }
}
"
`;

exports[`preset generator tsconfig.base.json should match snapshot 1`] = `
"{
  "compileOnSave": false,
  "compilerOptions": {
    "rootDir": ".",
    "moduleResolution": "Node16",
    "module": "Node16",
    "target": "ES2022",
    "lib": ["ES2022"],
    "esModuleInterop": true,
    "sourceMap": true,
    "declaration": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "noImplicitOverride": true,
    "noImplicitAny": false,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "noPropertyAccessFromIndexSignature": false,
    "noImplicitReturns": true,
    "noFallthroughCasesInSwitch": true,
    "importHelpers": true,
    "skipLibCheck": true,
    "skipDefaultLibCheck": true,
    "allowJs": true,
    "baseUrl": ".",
    "resolveJsonModule": true,
    "plugins": [
      // Transform paths in output .js files
      {
        "transform": "typescript-transform-paths"
      },
      // Transform paths in output .d.ts files (Include this line if you output declarations files)
      {
        "transform": "typescript-transform-paths",
        "afterDeclarations": true
      }
    ]
  },
  "exclude": ["node_modules", "tmp"]
}
"
`;
